app/resources/resources.py

```diff
@@ -0,0 +1,446 @@
+        except:
+            page = 1
+            limit = 20
+
+        # Get search parameter from request.
+        q = request.args.get('q')
+        if q is not None:
+            search = True
+
+        if bucket_id == 0:  # Bucket id is not specified in request
+            if search:  # Search for specific bucket list for user
+                record_query = BucketLists.query.filter_by(
+                    name=q, created_by=g.user.id).paginate(
+                    page, limit, False)
+            else:  # Query all bucket lists for user
+                record_query = BucketLists.query.filter_by(
+                    created_by=g.user.id).paginate(page, limit, False)
+
+            current_url = request.url
+            next_url, prev_url = "", ""
+
+            if record_query.has_next:
+                next_page = record_query.next_num
+                items = record_query.per_page
+                next_url = str(current_url[0:current_url.rindex('/')]) + \
+                    "?page=%s&limit=%s" % (str(next_page), str(items))
+
+            if record_query.has_prev:
+                prev_page = record_query.prev_num
+                items = record_query.per_page
+                prev_url = str(current_url[0:current_url.rindex('/')]) + \
+                    "?page=%s&limit=%s" % (str(prev_page), str(items))
+
+            # Serialize the query results in the JSON API format
+            results = bucket_list_schema.dump(
+                record_query.items, many=True).data
+
+            # results with paging
+            if len(results) > 0:
+                results = {"data": results, "paging":
+                           {"previous": prev_url, "next": next_url,
+                            "pageNum": page, "count": len(record_query.items)}}
+
+        else:  # Bucket id is specified in request
+            bucket_list_query = BucketLists.query.get_or_404(bucket_id)
+            if bucket_list_query.created_by == g.user.id:
+                # return results if they were created by user else return error
+                # message
+                results = bucket_list_schema.dump(
+                    bucket_list_query, many=False).data
+            else:
+                return {"message": "Unauthorized"}, 401
+            # Serialize the query results in the JSON API format
+
+        if len(results) > 0:
+            return results, 200
+        return "No data found", 404
+
+    @handle_exceptions
+    @auth.login_required
+    def post(self, bucket_id=0):
+        """
+        Creates bucketlists.
+
+        Args:
+            self, bucket_id
+
+        Return:
+            Creates a bucket for a logged in user
+        """
+
+        # parse incoming request data
+        parser = reqparse.RequestParser()
+        parser.add_argument('name')
+        args = parser.parse_args()
+        name = args['name']
+
+       # Validate the data or raise a Validation error if incorrect
```

Comment:
**Code Quality: Nitpick**

Loose the inline comments especially if the code is explicit.